File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        mp = {}                                          # unordered_map<int,int> mp
        v = []                                            # vector<int> v
        
        for i in range(len(nums)):                       # for(int i = 0; i < nums.size(); i++)
            rem = target - nums[i]                       # int rem = target - nums[i]
            
            if rem in mp:                                # if(mp.find(rem) != mp.end())
                v.append(mp[rem])                        # v.push_back(mp[rem])
                v.append(i)                              # v.push_back(i)
                break
            
            mp[nums[i]] = i                              # mp[nums[i]] = i (future use)
        
        return v

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        map<int,int>mp;
        int cnt = 0;
        int max_cnt = 0;
        for(int i = 0; i < nums.size(); i++)
        {
          mp[nums[i]]++;
        }
        for(int i = 0; i < nums.size(); i++)
        {
          if(mp.find(nums[i]-1) == mp.end()) 
          {
            while(mp.find(nums[i] + cnt) != mp.end())
            {
                cnt++;
            }
            max_cnt = max(cnt, max_cnt);
            cnt = 0;


          }
        }
        return max_cnt;
    }
};
